File: src/taskyn/web/frontend/src/api/timer.ts
import { api } from './client';
import type { ActiveTimer, TimeEntry, TimeEntryCreate } from '@/types';

export const timerApi = {
  start: (nodeId: string, notes?: string) =>
    api.post<TimeEntry>('/timer/start', { node_id: nodeId, notes }),

  stop: (entryId?: string) =>
    api.post<TimeEntry>('/timer/stop', entryId ? { entry_id: entryId } : {}),

  getCurrent: () =>
    api.get<ActiveTimer | null>('/timer/current'),

  getActive: () =>
    api.get<ActiveTimer[]>('/timer/active'),

  listTimeEntries: (projectId?: string) =>
    api.get<TimeEntry[]>(projectId ? `/time-entries?project_id=${projectId}` : '/time-entries'),

  logTime: (data: TimeEntryCreate) =>
    api.post<TimeEntry>('/time-entries', data),

  getTimeEntry: (id: string) =>
    api.get<TimeEntry>(`/time-entries/${id}`),

  deleteTimeEntry: (id: string) =>
    api.delete(`/time-entries/${id}`),
};

File: src/taskyn/web/backend/schemas/timer.py
"""Timer and time entry request schemas."""

from pydantic import BaseModel


class TimerStart(BaseModel):
    """Start timer request."""
    node_id: str
    notes: str | None = None


class TimerStop(BaseModel):
    """Stop timer request."""
    entry_id: str | None = None


class TimeEntryCreate(BaseModel):
    """Log time entry request."""
    node_id: str
    duration_minutes: int
    notes: str | None = None
